File: PasswordWallet.Client/src/app/mappers/folder.mapper.ts
import { FolderResponse } from '../contracts/responses/folder-response';
import { Folder } from '../models/folder';

export class FolderMapper {
  public static toModel = (source: FolderResponse): Folder => {
    return {
      id: source.id,
      name: source.name,
      position: source.position,
    };
  };
}

File: PasswordWallet.Server/src/Api/Endpoints/v1/Folders/FolderMapper.cs
using Api.Endpoints.v1.Folders.AddFolder;
using Core.Entities;
using FastEndpoints;

namespace Api.Endpoints.v1.Folders;

public class FolderMapper : ResponseMapper<FolderResponse, Folder>
{
    public override FolderResponse FromEntity(Folder folder) => new(folder.Id, folder.Name, folder.Position);

    public IEnumerable<FolderResponse> FromEntities(IEnumerable<Folder> folders) =>
        folders.Select(f => new FolderResponse(f.Id, f.Name, f.Position));

    public Folder ToEntity(AddFolderRequest req) => new()
    {
        Name = req.Name,
        UserId = req.UserId
    };
}